src/main/Whitelist.h

```diff
@@ -0,0 +1,38 @@
+#include "main/Application.h"
+#include "main/ManagedDataCache.h"
+#include "ledger/LedgerManager.h"
+
+namespace stellar
+{
+
+class Whitelist : public ManagedDataCache
+{
+  public:
+    static Whitelist* instance(Application& app);
+    size_t unwhitelistedReserve(size_t setSize);
+
+    bool isWhitelisted(std::vector<DecoratedSignature> signatures,
+                       Hash const& txHash);
+    bool isWhitelistSig(DecoratedSignature const& sig, Hash const& txHash);
+
+    int getLastUpdated(){
+        return lcl;
+    }
+
+    virtual std::string getAccount(Application& app) override;
+    virtual void fulfill(std::vector<DataFrame::pointer> dfs) override;
+
+  private:
+    Whitelist()
+    {
+        lcl = 0;
+    }
+    static Whitelist* mINSTANCE;
+
+    std::unordered_map<uint32_t, std::vector<string64>> whitelist;
+
+	// default to a 5% reserve
+	double mReserve = 0.05;
```

Comment:
It's a default.  We should be configuring the value on the account, even if it's the same, so that others will be able to query Horizon to find out what it is.  Even if "others" are just other node operators that can easily get the key for the whitelist account.